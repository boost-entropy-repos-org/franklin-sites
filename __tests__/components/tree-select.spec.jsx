import React from 'react';
import renderer from 'react-test-renderer';
import TreeSelect from '../../src/components/tree-select';
import treeData from '../../src/app/common/tree-data';

describe('TreeSelect component', () => {
  test('should render', () => {
    const component = renderer.create(<TreeSelect data={treeData} />).toJSON();
    expect(component).toMatchSnapshot();
  });

  test('should find the correct path', () => {
    const path = TreeSelect.getPath(treeData, 'Item 1b B');
    expect(path).toEqual([{ label: 'Item 1' }, { label: 'Item 1 b' }, { label: 'Item 1b B' }]);
  });

  test('should not find any paths', () => {
    const path = TreeSelect.getPath(treeData, 'The unfindable');
    expect(path).toBeUndefined();
  });
});
