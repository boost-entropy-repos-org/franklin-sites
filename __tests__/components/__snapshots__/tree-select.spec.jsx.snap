// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreeSelect component should render 1`] = `
<div
  className="tree-select"
>
  <button
    className="button dropdown"
    onClick={[Function]}
    type="button"
  >
    Select
  </button>
  <div
    className="tree-select-menu"
  >
    <ul
      className=""
    >
      <li
        className="branch"
      >
        <span
          className=""
          onClick={[Function]}
          role="button"
          tabIndex="0"
        >
          Item 1
        </span>
        <ul
          className=""
        >
          <li
            className=""
          >
            <span
              className=""
              onClick={[Function]}
              role="button"
              tabIndex="0"
            >
              Item 1 a
            </span>
          </li>
          <li
            className="branch"
          >
            <span
              className=""
              onClick={[Function]}
              role="button"
              tabIndex="0"
            >
              Item 1 b
            </span>
            <ul
              className=""
            >
              <li
                className=""
              >
                <span
                  className=""
                  onClick={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  Item 1b A
                </span>
              </li>
              <li
                className=""
              >
                <span
                  className=""
                  onClick={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  Item 1b B
                </span>
              </li>
            </ul>
          </li>
        </ul>
      </li>
      <li
        className=""
      >
        <span
          className=""
          onClick={[Function]}
          role="button"
          tabIndex="0"
        >
          Some Item 2
        </span>
      </li>
    </ul>
  </div>
</div>
`;
